api/utils/process.py

```diff
@@ -0,0 +1,138 @@
+
+
+async def process_patch_request(user, request, db_pool):
+    """Update user's dataset permissions."""
+    LOG.debug('Process PATCH request.')
+    # Put the JSON payload body into a dictionary object
+    request_body = await request.json()
+
+    # Take one connection from the active database connection pool
+    async with db_pool.acquire() as connection:
+        user_found = await user_exists(user, connection)
+        if user_found:
+            # user_identifier exists, continue with operations
+            # Check if user has permissions to be removed
+            permissions = await get_dataset_permissions(user, connection)
+            if permissions:
+                # Try to remove permissions before adding new permissions
+                await remove_dataset_permissions(user, connection)
+            # Parse datasets out of request_body
+            # ELIXIR API Specification has a typo in it, but this iterator satisfies it.
+            # Come back and fix this once the specification has been corrected
+            # dataset_group = [[p['affiliation'], p['datasets']] for p in body['permissions']]  # Correct form
+            dataset_group = [[p['affiliation'], p['datasets']] for p in request_body['datasets'][0]['permissions']]
+            if dataset_group:
+                # If any datasets were listed, add permissions for them
+                permissions_errors = await create_dataset_permissions(user, dataset_group, connection)
+                LOG.debug(f'Error list from creating permissions: {permissions_errors}. Should be empty!')
+                if permissions_errors:
+                    # If there are errors, return proper message
+                    return permissions_errors, True
+                else:
+                    # All inserts were processed successfully
+                    return None, True
+            else:
+                # Body contained no datasets, end operations here (permissions removed, none added)
+                return None, True
+        else:
+            return 'User not found', False
+
+
+async def process_delete_request(user, db_pool):
+    """Fetch user dataset permissions."""
+    LOG.debug('Process GET request.')
+
+    # Take one connection from the active database connection pool
+    async with db_pool.acquire() as connection:
+        user_found = await user_exists(user, connection)
+        if user_found:
+            # Check if user has permissions to be removed
+            permissions = await get_dataset_permissions(user, connection)
+            if permissions:
+                # Try to remove permissions before removing user
+                await remove_dataset_permissions(user, connection)
+            # Finally remove the user
+            user_removed = await delete_user(user, connection)
+            if user_removed:
+                return None, True
+            else:
+                return 'Database error when attempting to remove user', False
```

Comment:
suggestion of improvement/ try to see what happens: raise error here instead of returning a message and raising the error in the API function